packages/rs-drive/src/query/conditions.rs

```diff
@@ -1048,6 +1048,41 @@ impl<'a> WhereClause {
         Ok(query)
     }
 
+    /// Recursively find the property type for a given nested field name.
+    fn find_property_type(
+        document_type: &'a DocumentType,
+        field_name: &str,
+    ) -> Result<Cow<'a, DocumentPropertyType>, Error> {
+        let parts = field_name.split('.');
+
+        let mut current_properties = document_type.properties();
+        let mut current_property_type = None;
+
+        for part in parts {
```

Comment:
looks a bit as use case for `Iterator::fold`

Reply:
I changed to use fold. I think it's harder to understand now. But it works the same. Let me know what you think.

Reply:
Reverted